db.go

```diff
@@ -152,7 +157,7 @@ func (x *dummyUserStore) Close() {
 func (x *dummyUserStore) SetPassword(userId UserId, password string) error {
 	x.usersLock.Lock()
 	defer x.usersLock.Unlock()
-	if user, exists := x.users[userId]; exists {
+	if user, exists := x.users[userId]; exists && user.authUserType == IMQSUserType {
```

Comment:
Rather create a method `(u AuthUserType) CanSetPassword() bool` and use that method here and wherever else the logic applies.